osutil/vfs/lists/lists.go

```diff
@@ -88,21 +92,54 @@ func (n *Node[T]) linkBefore(other *Node[T]) {
 	prev.next = other
 }
 
+// NodePointerer is an interface that types must implement to provide access
+// to their embedded Node[T].
+//
+// This is used by [containedNode] to create a NodeContainer[T] from a T.
+type NodePointerer[T any] interface {
+	NodePointer(*T) *Node[T]
+}
+
+// containedNode is a private type that wraps a node embedded in a structure
+// of type T. It is used to provide type safety for list operations, by ensuring
+// that node container pointer was set correctly.
+type containedNode[T any] struct {
+	n *Node[T]
+}
+
+// ContainedNode returns a private that ensures node container pointer is initialized.
+//
+// The returned value is suitable for use with List.Append and List.Prepend.
+func ContainedNode[NP NodePointerer[T], T any](c *T) containedNode[T] {
```

Comment:
is this going to be used a lot later? it could have  short name if that's the case, otherwise it's ok

Reply:
It's going to be used a few times. I may tweak the name to something shorter before merging. It's mostly used in tests, in code I expect less than a dozen lines in total later on. Right now it's only one.